test/hotspot/jtreg/compiler/vectorization/TestFloat16VectorOperations.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2025, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+* @test
+* @bug 8346236
+* @summary Auto-vectorization support for various Float16 operations
+* @requires vm.compiler2.enabled
+* @modules jdk.incubator.vector
+* @library /test/lib /
+* @compile TestFloat16VectorOperations.java
+* @run driver compiler.vectorization.TestFloat16VectorOperations
+*/
+
+package compiler.vectorization;
+import compiler.lib.ir_framework.*;
+import jdk.incubator.vector.Float16;
+import static jdk.incubator.vector.Float16.*;
+import static java.lang.Float.*;
+import jdk.test.lib.*;
+import compiler.lib.generators.*;
+
+public class TestFloat16VectorOperations {
+    private short[] input1;
+    private short[] input2;
+    private short[] input3;
+    private short[] output;
+    private static short SCALAR_FP16 = (short)0x7777;
+    private static final int LEN = 2048;
+
+    public static void main(String args[]) {
+        TestFramework.runWithFlags("-XX:-TieredCompilation", "-Xbatch","--add-modules=jdk.incubator.vector");
+    }
+
+    public static boolean assertResults(short expected, short actual) {
+        Float16 expected_fp16 = valueOf(expected);
+        Float16 actual_fp16 = valueOf(actual);
+        if (isNaN(expected_fp16) ^ isNaN(actual_fp16)) {
+            return !expected_fp16.equals(actual_fp16);
+        }
+        return false;
```

Comment:
This should be reverse:
        if (isNaN(expected_fp16) ^ isNaN(actual_fp16)) {
          return false;
        }
        return !expected_fp16.equals(actual_fp16);

Reply:
My bad. Fixed.